xskillscore/core/np_deterministic.py

```diff
@@ -28,19 +41,38 @@ def _pearson_r(a, b, axis):
     scipy.stats.pearsonr
 
     """
+    weights = _check_weights(weights)
     a = np.rollaxis(a, axis)
     b = np.rollaxis(b, axis)
-    ma = np.mean(a, axis=0)
-    mb = np.mean(b, axis=0)
+
+    # Only do weighted sums if there are weights. Cannot have a
+    # single generic function with weights of all ones, because
+    # the denominator gets inflated when there are masked regions.
+    if weights is not None:
```

Comment:
I'm not thrilled with this solution in terms of ballooning the code, but I don't see a better route. I couldn't find a syntax where one could do something like `a * weights, ...)` and have weights be `None`, `[]`, etc. to be ignored. As mentioned in https://github.com/raybellwaves/xskillscore/issues/30, we can't just pass an array of 1's since a spatially masked domain breaks here. It's better to just not weight whatsoever (i.e. follow the current `xskillscore` master branch).

Reply:
I agree with this approach. I thing we should err on the side of ease of implementation over code length. I also mentioned this in https://github.com/raybellwaves/xskillscore/issues/32#issuecomment-544211745